### PR TITLE
Product filter only matches if name begins withs search term

### PR DESCRIPTION
**Pull request by rhodri**
Commit to address #1233

### CODE REVIEW

**Comment by lin-d-hop** [comment]
Hi @rhodri 
The build is failing on this. Could you take a look before I more for code review?
Cheers!

**Comment by rhodri** [comment]
ok, looks like a failing spec. will try and add a few while I'm at it.

File: app/assets/javascripts/darkswarm/services/matcher.js.coffee [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
     match: (properties, text)->
       properties.some (prop)->
         prop ||= ""
         prop.toLowerCase().indexOf(text.toLowerCase()) != -1
+
+    # Return true if text occurs at the beginning of any word present in an array of strings
+    matchBeginning: (properties, text) ->
+      text = text.trim()
+      properties.some (prop) ->
+        prop ||= ""
```
Comment (oeoeaio) [inline app/assets/javascripts/darkswarm/services/matcher.js.coffee]: What about a regular expression here to do a case-insensitive search for ``text`` at the beginning of the prop name or following a whitespace character or dash (or whatever other characters we choose to specify). 

Something like (untested):
```
text = text.trim()
regexp = new RegExp("(?:^|[\\s-])#{text}", "i")
properties.some (prop) -> (prop || "").search(regexp) >= 0
```

**Comment by lin-d-hop** [comment]
Hiya @rhodri Have you had a chance to look at this?
Let me know if I can do anything to help :-)

**Comment by daniellemoorhead** [comment]
---Doing a PR Cleanup---

@lin-d-hop thoughts on what next for this?

**Comment by daniellemoorhead** [comment]
@mkllnk @lin-d-hop given #1545 fixed this issue and was merged should this older PR be closed off so it's not hanging around anymore?

**Pull request closed without merging**